src/document.cpp

```diff
@@ -620,6 +641,29 @@ rvoe<NoReturnValue> PdfDocument::create_catalog() {
   /Pages {} 0 R
 )",
                    pages_object);
+
+    if(!page_labels.empty()) {
+        buf += R"(  /PageLabels
+    << /Nums [
+)";
+        for(const auto &page_label : page_labels) {
+            std::format_to(app, R"(      {} <<)", page_label.start_page);
```

Comment:
All of these format strings should all end with a `\n` and have the requisite amount of indentation spaces. Each of them should be on a separate line, not just separated with a single space.

Reply:
This formatting is very ad-hoc in capypdf. Structurally you are likely to have these errors a lot.

Reply:
That is unfortunately quite true. Ideally for formatting you'd want to keep the PDF objects stored in some sort of PDF structured format that can be programmatically written to disk. That is terrible for development, though, having them in "human meaningful" structs etc is a lot better in terms of development and especially debugging. So then it is a question of either converting those to a PDF object format entities first and then dumping that out or writing the file in an ad-hoc fashion. The former requires that you write _that_ conversion code by hand and then if you are _already_ doing that, you might as well dump them out directly to the output file.